lib/ensemble/test_format.py

```diff
@@ -0,0 +1,167 @@
+import unittest
+
+from format import _is_dict_w_strings
+from format import _is_point
+from format import Chart
+from format import Report
+
+class IsFunctionTests(unittest.TestCase):
+    def test_is_dict_w_strings(self):
+        # Not a dict
+        self.assertFalse(_is_dict_w_strings("foo", None))
+
+        # Dict w keys too short
+        value = {"a": "a", "b": "b", "c": "c"}
+        self.assertFalse(_is_dict_w_strings(value, ["a", "b"]))
+
+        # Dict w keys too long
+        value = {"a": "a", "b": "b"}
+        self.assertFalse(_is_dict_w_strings(value, ["a", "b", "c"]))
+
+        # Dict w wrong keys
+        self.assertFalse(_is_dict_w_strings(value, ["b", "c"]))
+
+        # Dict w matching keys
+        self.assertTrue(_is_dict_w_strings(value, ["a", "b"]))
+
+        # Dict w matching keys different order
+        self.assertTrue(_is_dict_w_strings(value, ["b", "a"]))
+
+        # Dict w matching keys, values not strings
+        value = {"a": 1, "b": 2}
+        self.assertFalse(_is_dict_w_strings(value, ["a", "b"]))
+
+    def test_is_point(self):
+        # Not a tuple
+        self.assertFalse(_is_point([1, 2]))
+
+        # Not numbers
+        self.assertFalse(_is_point(("a", "b")))
+
+        # Wrong length
+        self.assertFalse(_is_point((1, 2, 3)))
+
+        # Just right
+        self.assertTrue(_is_point((1, 2)))
+
+
+class ChartTests(unittest.TestCase):
+    output = '''{"data": [{"x": 256, "y": 43.8}, {"x": 512, "y": 21.2}, {"x": 1024, "y": 31.4}, {"x": 2048, "y": 3.6}], "description": "The amount of memory the graphics card has", "labels": {"x": "Memory", "y": ""}, "section": "gfx", "title": "Memory", "units": {"x": "MB", "y": "%"}}'''
+
+    def test_init_failures(self):
+        # string args aren't strings
+        with self.assertRaises(TypeError):
+            chart = Chart(1, "title", "desc", "section")
```

Comment:
Since you never use the `chart` variable you can just call `Chart` and not assign it to anything.